### PR TITLE
remove serde.retention, use retention.max-hr * hour_seconds 

### PR DESCRIPTION
**Pull request by chenqin**
- group max-mb max-hr under retention configuration
- remove ser.retention use max-hr double value multiple with 3600 as kafka ingestion retention config

### CODE REVIEW

File: src/meta/ClusterInfo.cpp [hunk 1]
```diff
@@ -256,14 +256,11 @@ unordered_map<std::string, Column> asColumnProps(const YAML::Node& node) {
...
-KafkaSerde asSerde(const YAML::Node& node) {
+KafkaSerde asSerde(const YAML::Node& node, const double retention) {
   KafkaSerde serde;
   if (node) {
     serde.protocol = node["protocol"].as<std::string>();
-    auto retention = node["retention"];
-    if (retention) {
-      serde.retention = retention.as<uint64_t>();
-    }
+    serde.retention = (uint64_t) retention;
```
Comment (shawncao) [inline src/meta/ClusterInfo.cpp:263]: This retention is KAFKA topic retention in Kafka store, not the max-hr (retention in Nebula), the usage of this value is used to estimate offset and range for each Kafka split when early version of Kafka doesn't have watermark/offset feature, we should leave this as it is.

File: src/meta/ClusterInfo.cpp [hunk 2]
```diff
@@ -356,18 +353,21 @@ void ClusterInfo::load(const std::string& file, CreateMetaDB createDb) {
       name = td["topic"].as<std::string>();
     }
 
+    // hour table level retention.max-hr as single way to ingest and evict data
+    const auto retentionSeconds = (td["retention"]["max-hr"].as<double>() * Evidence::HOUR_SECONDS);
```
Comment (shawncao) [inline src/meta/ClusterInfo.cpp:357]: nit: can we use a temporary var to capture td["retention"] for reuse?

**Review by shawncao — CHANGES_REQUESTED** [review]
I think the change in for Kafka serde is inappropriate, otherwise looks good!

File: src/meta/ClusterInfo.cpp (hunk 1 above)
Comment (chenqin) [reply src/meta/ClusterInfo.cpp:263]: got it, how about rename to topic-retention to avoid confusion.

File: src/meta/ClusterInfo.cpp (hunk 2 above)
Comment (chenqin) [reply src/meta/ClusterInfo.cpp:357]: yes,

**Review by shawncao — APPROVED** [review]
sounds good to me, thanks for the update.

**Comment by chenqin** [comment]
Notice to upgrade table compatible with this change
max-mb -> retention.max-mb
max-hr -> retention.max-hr
serde.retention -> serde.topic-retention

**Pull request merged**